src/main/java/internet/store/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,195 @@
+package internet.store.dao.jdbc;
+
+import internet.store.dao.UserDao;
+import internet.store.exception.DataProcessingException;
+import internet.store.lib.Dao;
+import internet.store.model.Role;
+import internet.store.model.User;
+import internet.store.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Dao
+public class UserDaoJdbcImpl implements UserDao {
+    @Override
+    public Optional<User> findByLogin(String login) {
+        User user = getUserByLogin(login);
+        user.setRoles(getUserRoles(user.getUserId()));
+        return Optional.of(user);
```

Comment:
is user is null you will have npe on user.setRoles(), let's add check before that